yarpl/test/FlowableFlatMapTest.cpp

```diff
@@ -271,5 +271,44 @@ TEST(FlowableFlatMapTest, MultithreadedLargeAmount) {
   p2->evb.stop();
 }
 
+TEST(FlowableFlatMapTest, MergeTest) {
+  auto sub = yarpl::make_ref<TestSubscriber<std::string>>(0);
+
+  auto p1 = Flowables::justN<std::string>({"foo", "bar"});
```

Comment:
It is possible to write a test like this
 - One flowable from Thread A
 - One flowable from Thread B
 - They are merged into another flowable which lives in Thread C
So it will be a total merge of the flowables from different threads into a single thread.

Reply:
Tests in this compilation unit do exercise multiple independent threads pushing into flatMap (and therefore merge, as merge is just a very light wrapper around flatMap)